### PR TITLE
feat(replays): Add killswitch for mobile replays

### PR DESCRIPTION
**Pull request by cmanallen**
Organizations are not allowed unless the `organizations:session-replay-video` flag evaluates to `true`.

Related: https://github.com/getsentry/sentry/issues/63255

Related PRs:
https://github.com/getsentry/getsentry/pull/13596/files
https://github.com/getsentry/sentry/pull/68567/files

### CODE REVIEW

**Review by jjbayer — APPROVED** [review]
Code looks good to me, some test coverage would be nice.

File: relay-dynamic-config/src/feature.rs [hunk 1]
```diff
@@ -10,6 +10,11 @@ pub enum Feature {
     /// Serialized as `organizations:session-replay`.
     #[serde(rename = "organizations:session-replay")]
     SessionReplay,
+    /// Enables ingestion of mobile Session Replays.
+    ///
+    /// Serialized as `organizations:session-replay-mobile`.
+    #[serde(rename = "organizations:session-replay-mobile")]
+    SessionReplayMobile,
```
Comment (iker-barriocanal) [inline relay-dynamic-config/src/feature.rs]: Is it `replay.organizations.mobile-slug-denylist` (from the PR description) or `organizations:session-replay-mobile`? I couldn't find any matches in (get)sentry.

**Review by iker-barriocanal — APPROVED** [review]
Approving not to block on the option name, but please align the PR description with the code before merging.

File: relay-server/src/services/processor/replay.rs [hunk 2]
```diff
@@ -88,6 +90,8 @@ pub fn process(
                 item.set_payload(ContentType::OctetStream, replay_recording);
             }
             ItemType::ReplayVideo => {
+                contains_mobile_event = true;
```
Comment (Dav1dde) [inline relay-server/src/services/processor/replay.rs]: I assume `Video == Mobile`, but can't we name the feature that way too, we just have a feature that toggles the videos? Just from reading the feature I wouldn't have expected this.

File: relay-server/src/services/processor/replay.rs [hunk 3]
```diff
@@ -105,6 +109,13 @@ pub fn process(
         }
     }
 
+    // If the envelope contained a mobile event and mobile replay is disabled drop
+    // the envelope.
+    if contains_mobile_event && !mobile_replay_enabled {
+        state.managed_envelope.drop_items_silently();
+        return Ok(());
+    }
```
Comment (Dav1dde) [inline relay-server/src/services/processor/replay.rs]: We fully process the envelope before quitting, can we not quit out early if the feature flag is not enabled?

File: relay-dynamic-config/src/feature.rs (hunk 1 above)
Comment (cmanallen) [reply relay-dynamic-config/src/feature.rs]: Renamed to `organizations:session-replay-video`.

File: relay-server/src/services/processor/replay.rs (hunk 3 above)
Comment (cmanallen) [reply relay-server/src/services/processor/replay.rs]: Added early exit.

File: relay-server/src/services/processor/replay.rs (hunk 2 above)
Comment (cmanallen) [reply relay-server/src/services/processor/replay.rs]: Renamed to "video".

File: CHANGELOG.md [hunk 4]
```diff
@@ -9,6 +9,8 @@
 
 **Features**:
 
+- Add feature flag for replay video event types. ([#3402](https://github.com/getsentry/relay/pull/3402))
+
```
Comment (Dav1dde) [inline CHANGELOG.md]: When you merge/rebase please move this under the `Breaking Changes`.

**Review by Dav1dde — APPROVED** [review]

File: relay-server/src/services/processor/replay.rs [hunk 5]
```diff
@@ -88,6 +89,13 @@ pub fn process(
                 item.set_payload(ContentType::OctetStream, replay_recording);
             }
             ItemType::ReplayVideo => {
+                // If video replays have not been enabled and a video replay was received
+                // drop the envelope.
+                if !video_replay_enabled {
+                    state.managed_envelope.drop_items_silently();
+                    return Ok(());
+                }
```
Comment (iker-barriocanal) [inline relay-server/src/services/processor/replay.rs]: nit: the comment is not helpful as it describes what the `match` variant and `if` condition match on, so I'd suggest deleting it.

**Review by iker-barriocanal — APPROVED** [review]

**Pull request merged**